Yoast/YoastSEO.js
# Add query string appender and use it in the assessments

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Added a QueryStringAppender class.
* Created a singleton instance wrapper for QueryStringAppender that puts itself on the global. The singleton is the pragmatic choice so that we do not have to rework all the older style assessments to listen to configuration right now. The adding to the global is because the custom assessments live in a separate environment and we do want them to be able use the same instance.
* Had to refactor the worker's initialize function a bit because of ESLint max statements error.

## Test instructions

This PR can be tested by following these steps:

* Check all the assessments to see if the links contain the extra parameters.
* Check on post/page and taxonomy.

For an overview of the altered files:
[Readability](https://github.com/Yoast/YoastSEO.js/wiki/Scoring-readability-analysis):
- Flesch Reading Ease
- Paragraph length
- Passive voice
- Consecutive sentences
- Sentence length
- Subheading distribution
- Not enough content
- Transition words

[SEO](https://github.com/Yoast/YoastSEO.js/wiki/Scoring-SEO-analysis):
- Internal links
- Keyphrase in introduction
- Keyphrase distribution
- Keyphrase length
- Keyphrase density
- Stopwords
- Keyphrase in meta description
- Meta description length
- Outbound links
- SEO title width
- Keyphrase in subheading
- Text length (Taxonomy)
- Link keyphrase
- Image alt attributes
- Text length
- Keyphrase in title
- Keyphrase in slug
- Slug too long
- Slug stopwords
- Previously used keyphrase

Fixes #1876
